glib/src/subclass/object.rs

```diff
@@ -224,54 +224,39 @@ mod test {
     use super::super::super::value::{ToValue, Value};
     use super::*;
     use crate::{StaticType, Type};
+    use crate as glib;
 
     use std::cell::RefCell;
 
     mod imp {
         use super::*;
 
         // A dummy `Object` to test setting an `Object` property and returning an `Object` in signals
+        #[derive(Default)]
         pub struct ChildObject;
+
+        #[glib::object_subclass]
         impl ObjectSubclass for ChildObject {
             const NAME: &'static str = "ChildObject";
```

Comment:
I wonder if we could have a default for the `NAME` as well?

Reply:
That default should probably include the crate and module name though

Reply:
Oh, I hadn't even thought of that. It would be great, but I don't see a way for a proc macro to get the equivalent of `std::module_path!()` (which it could manipulate to generate a name). `Span` can get the source file path (with unstable methods), but I don't see anything that could really be used for this.

It can be added later though, if we find a way or Rust adds one.

Reply:
Would things like `crate::foo::bar::SomeObject` work and be reasonable as a default value for `NAME`? I think that could be done by having the proc macro generate `concat!(std::module_path!(), "::", type_name)`. But if it had to remove/replace the `::` in the module path, it couldn't do that at compile time.

Reply:
I don't think that's a valid GType name

Reply:
[It appears so](https://developer.gnome.org/gobject/stable/gtype-conventions.html):
>Type names (including object names) must be at least three characters long and start with ‘a–z’, ‘A–Z’ or ‘_’.

Reply:
So probably not really possible to provide a default for `NAME` now, but it would be great to add later if someone finds a way to provide a good default.

Reply:
Or go with just the type name. At least for applications that's often going to be fine

Reply:
Often, but it doesn't seem great to me to end up with a crash at run time when a user uses too generic a name for their type (I assume that's what would happen), which isn't normally a concern in Rust.

Reply:
It would cause an assertion, yes. I agree